Queue Is Implemented

urvashi-code1255 opened the pull request:
# Description

Please include a summary of the change and which issue is fixed. List any dependencies that are required for this change.
Fixed #15 
=> I have Implemented Enqueue, Dequeue, front and Display function of queue.

## Type of change

Please check options that are relevant to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New Data Structure 
- [ ] New Algorithm

# Checklist:

- [x] I have squashed my commits
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My code is a working bug-free code

Commit:
Queue Implementation Added

Commit:
Delete Queue_Implementation.exe

Commit:
added Queues

ay2306 commented on DS implementations/QUEUES/Queue_Implementation.cpp:
```diff
@@ -0,0 +1,138 @@
+        // we haven't fill any element in the queue yet so rear and front both are at starting index.
+        front = rear = 0;
+        capacity = c;
+        queue = new int;
```
declare size as well

ay2306 commented on DS implementations/QUEUES/Queue_Implementation.cpp:
```diff
@@ -0,0 +1,138 @@
+    }
+
+    // function to insert an element at the rear of the queue
+    void queueEnqueue(int data)
```
keep the method name as push

ay2306 replied:
Handle condition for Queue Overflow

ay2306 commented on DS implementations/QUEUES/Queue_Implementation.cpp:
```diff
@@ -0,0 +1,138 @@
+        // check queue is full or not
+        if (capacity == rear)
+        {
+            printf("\nQueue is full\n");
```
The queue is empty in this case

ay2306 commented on DS implementations/QUEUES/Queue_Implementation.cpp:
```diff
@@ -0,0 +1,138 @@
+    }
+
+    // function to delete an element from the front of the queue
+    void queueDequeue()
```
keep the method name as pop

ay2306 replied:
The pop operation should be of time complexity `O(1)`, you implemented `O(n)` operation

ay2306 commented on DS implementations/QUEUES/Queue_Implementation.cpp:
```diff
@@ -0,0 +1,138 @@
+    }
+
+    // print front of queue
+    void queueFront()
```
keep the method name just as `front()`

Commit:
Queue is updated

Commit:
Documentation updated

urvashi-code1255 commented:
Declared Size variable , as in queue we use enqueue and dequeue so i have changed the name to enqueue, dequeue, front, rear.
I have handled the overflow and underflow conditions. Also, I have performed the dequeue functionn in O(1) which was implemented in O(n) before  and I have added the comments for every single difficult line to make it easily understandable

ay2306 commented on DS implementations/QUEUES/Queue_Implementation.cpp:
```diff
@@ -0,0 +1,108 @@
+    // assigning last index of the queue to the rear.
+    queue->rear = capacity - 1;
+    // dynamically allocating int array of capacity.
+    queue->array = new int[(queue->capacity * sizeof(int))]; 
```
You do not need to multiply with the size of datatype using `new` keyword
You are confused with `malloc` here

ay2306 commented on DS implementations/QUEUES/Queue_Implementation.cpp:
```diff
@@ -0,0 +1,108 @@
+// C++ program for array implementation of queue
+
+#include <bits/stdc++.h>
```
Either do no include `bits/stdc++.h` or do not `use namespace std` otherwise there will ambiguity with container `queue` from STL

ay2306 commented on DS implementations/QUEUES/Queue_Implementation.cpp:
```diff
@@ -0,0 +1,108 @@
+}
+
+// Function to remove an item from queue. It changes front and size
+int dequeue(Queue *queue)
```
the variable name is not suitable. A simple `q`, `que` will be better. Even if code is compiling and running due to the scope of the variable. This will create a lot of confusion.

ay2306 reviewed (changes requested):
Great code. There are some little issues which I have mentioned. Please make the requested changes.

Commit:
name of the queue changed to q

ay2306 committed:
Merge branch 'master' into master

urvashi-code1255 commented:
There are no conflicts now.
